lib/content/mixin.js

```diff
@@ -130,8 +130,24 @@ export default {
       type: Boolean,
       default: false,
     },
+    timeSpent: {
+      type: Number,
+      default: null,
```

Comment:
Looking at the implementation, maybe for simplicity we should make this default to `0`? That way the `durationBasedProgress` will just be `0` when `timeSpent` is not specified.